File: db/src/main/java/com/delivery/db/repository/UserOrderMenuRepository.java
package com.delivery.db.repository;

import com.delivery.db.entity.UserOrderMenu;
import com.delivery.db.enums.UserOrderMenuStatus;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.List;

public interface UserOrderMenuRepository extends JpaRepository<UserOrderMenu, Long> {
    List<UserOrderMenu> findAllByUserOrderIdAndStatusOrderById(Long userOrderId, UserOrderMenuStatus status);
}

File: db/src/main/kotlin/com/delivery/db/repository/UserOrderMenuRepository.kt
package com.delivery.db.repository

import com.delivery.db.entity.UserOrderMenu
import com.delivery.db.enums.UserOrderMenuStatus
import org.springframework.data.jpa.repository.JpaRepository

interface UserOrderMenuRepository : JpaRepository<UserOrderMenu, Long> {
    fun findAllByUserOrderIdAndStatusOrderById(userOrderId: Long?, status: UserOrderMenuStatus?): List<UserOrderMenu>
}